### PR TITLE
Compute 2020-06-01 API version

### PR DESCRIPTION
**Pull request by hyonholee**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by kamusta-msft — APPROVED** [review]

**Review by hari-bodicherla — APPROVED** [review]

**Review by sirajudeens — APPROVED** [review]

**Review by mirichmo — APPROVED** [review]

**Review by dochung4 — APPROVED** [review]

**Review by avinashakka — APPROVED** [review]

**Review by jemex — APPROVED** [review]

**Comment by hyonholee** [comment]
Compute team signed off this PR.  ARM team signed off all PRs into this branch.

**Comment by chiragg4u** [comment]
>       "post": {

Why is this operation modeled as POST action?

---
Refers to: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json:3441 in f1abbbc. [](commit_id = f1abbbc485893ba79b1dd47f866ddf39fb15e953, deletion_comment = False)

**Comment by chiragg4u** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/retrieveBootDiagnosticsData": {

Can this be just bootDiagnostics?

---
Refers to: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json:3440 in f1abbbc. [](commit_id = f1abbbc485893ba79b1dd47f866ddf39fb15e953, deletion_comment = False)

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/examples/VirtualMachineRunCommand.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "subscriptionId": "24fb23e3-6ba3-41f0-9b6e-e41131d5d61e",
+    "resourceGroupName": "crptestar98131",
+    "vmName": "vm3036",
+    "$top": 1,
+    "api-version": "2020-06-01",
+    "monitor": "true",
+    "parameters": {
+      "commandId": "RunPowerShellScript"
+    }
+  },
```
Comment (chiragg4u) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/examples/VirtualMachineRunCommand.json:12]: Please add valid examples here.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+        }
+      },
+      "description": "Instance view status."
+    },
+    "RunCommandResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InstanceViewStatus"
```
Comment (chiragg4u) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json:444]: >InstanceViewStatus [](start = 35, length = 18)

Why is this named InstanceViewStatus?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmScaleSetName}/virtualmachines/{instanceId}/runCommand": {
+      "post": {
+        "tags": [
```
Comment (chiragg4u) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json:187]: Please add example of long running RunCommand and how will the user be able to get the updated status.

**Review by chiragg4u — APPROVED** [review]
I've added few comments, Please resolve, I'll mark ARMApproved.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json (hunk 3 above)
Comment (koralski) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json:187]: I am adding the missing example with PR: https://github.com/Azure/azure-rest-api-specs/pull/9875

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json (hunk 2 above)
Comment (koralski) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/runCommands.json:444]: These corresponds to a public class: InstanceViewStatus
https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.compute.models.instanceviewstatus?view=azure-dotnet

**Review by chiragg4u — APPROVED** [review]
Approved. Please resolve SDK failures.

**Comment by NullMDR** [comment]
Hi @hyonholee The autorest reports an error:

ERROR (PreCheck/DuplicateSchema): Duplicate Schema named UpdateResource -- properties.id: undefined => {"description":"Resource Id","readOnly":true,"$ref":"#/components/schemas/schemas:1026"},properties.name: undefined => {"description":"Resource name","readOnly":true,"$ref":"#/components/schemas/schemas:1027"},properties.type: undefined => {"description":"Resource type","readOnly":true,"$ref":"#/components/schemas/schemas:1028"} 

Could you please fix it?

**Comment by hyonholee** [comment]
That error is for package-2019-07-01-only tag.  Why this failure happens suddenly?  This PR does not change anything for 2019-07-01 version.  Only adds 2020-06-01 version.

##[error]2020-06-19T17:59:41.522Z: [Azure/azure-sdk-for-python][AutoRest] Failure during batch task - {"tag":"package-2019-07-01-only"} -- Error: Plugin prechecker reported failure..'

**Review by PhoenixHe-NV — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**